src/trio/_subprocess.py

```diff
@@ -760,6 +760,9 @@ async def read_output(
             async for chunk in stream:
                 chunks.append(chunk)
 
+    # TODO: if this nursery raises an exceptiongroup, manually raise a CalledProcessError,
+    # or maybe a InternalTrioError, with the exceptiongroup as a cause, so run_process
+    # never raises an exceptiongroup. (?)
     async with trio.open_nursery() as nursery:
         # options needs a complex TypedDict. The overload error only occurs on Unix.
         proc = await open_process(command, **options)  # type: ignore[arg-type, call-overload, unused-ignore]
```

Comment:
.

Reply:
(surfacing from an ~hour-long dive into `run_process()`)  I think it's sufficiently unlikely to get an `ExceptionGroup` here from any reasonable usage, that I'd favor `raise InternalTrioError from the_group`.

Reply:
Getting multiple exceptions from inside the nursery may be unlikely, but it seems relatively straightforward to pass a bad command that will get an exception from `open_process` - and we probably don't want to raise `InternalTrioError` in those cases. Pushing a commit with a suggested fix, but I suspect there might be a reason the `open_process` call was placed inside the nursery in the first place (?).

Reply:
(the `PermissionError` in the test is raised inside `open_process`, and there's validation of inputs that `run_process` punts to `open_process` as well)